test/integration/oauth_token_endpoint_test.go

```diff
@@ -0,0 +1,139 @@
+	}
+
+	anonConfig := restclient.Config{}
+	anonConfig.Host = clientConfig.Host
+	anonConfig.CAFile = clientConfig.CAFile
+	anonConfig.CAData = clientConfig.CAData
+	transport, err := restclient.TransportFor(&anonConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Hit the token request endpoint
+	masterURL, err := url.Parse(clientConfig.Host)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	masterURL.Path = "/oauth/token/request"
+
+	response, tokenHeaderLocation := checkNewReqAndRoundTrip(t, transport, masterURL.String(), false, http.StatusFound)
+	response.Body.Close()
+
+	if len(tokenHeaderLocation) == 0 {
+		t.Fatalf("no Location header")
+	}
+
+	authRedirect, err := url.Parse(tokenHeaderLocation)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	response, authHeaderLocation := checkNewReqAndRoundTrip(t, transport, authRedirect.String(), true, http.StatusFound)
+	response.Body.Close()
+
+	if len(authHeaderLocation) == 0 {
+		t.Fatalf("no Location header")
+	}
+
+	displayResp, _ := checkNewReqAndRoundTrip(t, transport, authHeaderLocation, false, http.StatusOK)
+	apiToken := getTokenFromDisplay(displayResp)
+	if apiToken == "" {
+		t.Fatalf("API Token not found in display")
+	}
+	displayResp.Body.Close()
+
+	// Verify use of the bearer token
+	userConfig := &anonConfig
+	userConfig.BearerToken = apiToken
+	userClient, err := userclient.NewForConfig(userConfig)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	userWhoamiOptions := cmd.WhoAmIOptions{UserInterface: userClient.Users(), Out: ioutil.Discard}
+	retrievedUser, err := userWhoamiOptions.WhoAmI()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if retrievedUser.Name != "foo" {
+		t.Fatalf("got %v, expected foo", retrievedUser.Name)
+	}
+}
+
+// Parse the HTML body for the API token contained in the first <code></code> block
+func getTokenFromDisplay(response *http.Response) string {
+	tokenizer := html.NewTokenizer(response.Body)
+
+	var seenCode bool
+	for tokenType := tokenizer.Next(); tokenType != html.ErrorToken; {
+		token := tokenizer.Token()
+		if tokenType == html.StartTagToken && token.Data == "code" {
+			seenCode = true
+		}
+		if seenCode && tokenType == html.TextToken {
+			return token.Data
+		}
+		tokenType = tokenizer.Next()
+	}
+
+	return ""
+}
+
+func checkNewReqAndRoundTrip(t *testing.T, rt http.RoundTripper, url string, doBasicAuth bool, expectedCode int) (*http.Response, string) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	req.Header.Set("Accept", "text/html; charset=UTF-8")
+
+	if doBasicAuth {
+		req.SetBasicAuth("foo", "bar")
+	}
+
+	resp, err := rt.RoundTrip(req)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	if resp.StatusCode != expectedCode {
+		t.Fatalf("unexpected response code %v", resp.StatusCode)
+	}
+
+	return resp, resp.Header.Get("Location")
```

Comment:
return location and the body's bytes and close the body in this function always.